homeassistant/components/prometheus/__init__.py

```diff
@@ -318,6 +323,32 @@ def _handle_climate(self, state):
                     float(action == current_action)
                 )
 
+    def _handle_humidifier(self, state):
+        humidity = state.attributes.get(ATTR_HUMIDITY)
+        if humidity:
+            metric = self._metric(
+                "humidity", self.prometheus_cli.Gauge, "Target Relative Humidity"
+            )
+            metric.labels(**self._labels(state)).set(humidity)
+
+        metric = self._metric(
+            "humidifier_state",
+            self.prometheus_cli.Gauge,
+            "State of the humidifier (0/1)",
+        )
+        try:
+            value = self.state_as_number(state)
+        except ValueError:
+            value = 0.0
```

Comment:
0 should mean the off state, please don't just export it if there's an error. Exporting no state metric would be more appropriate.